assets/js/components/notifications/BannerNotification/index.js

```diff
@@ -123,6 +124,10 @@ function BannerNotification( {
 			// Set the dismissed flag in cache without immediately hiding it.
 			await persistDismissal();
 		}
+
+		if ( onView ) {
+			await onView();
+		}
```

Comment:
@asvinb, I know that it was defined in IB, but why do we need this `onView` property in this banner component? Why not to use `useMount` hooks in `GatheringDataNotification` and `ZeroDataNotification` components instead of `handleOnView` callback hooks? I think if we remove this `onView` property and rework notification components, then we will have to support less code, right?

Reply:
@eugene-manuilov So the original IB suggested not to break down the component into 2 individual ones. Rather have a single one and thus the `onView` prop would have a callback for each of the 2 `BannerNotification` components within the same file (`ZeroDataStateNotifications`). Context: https://github.com/google/site-kit-wp/issues/5042#issuecomment-1091672724
@felixarntz then suggested to still break down the components and he was fine with keeping `onView`. https://github.com/google/site-kit-wp/issues/5042#issuecomment-1092158463 But yeah, looking at your comment, it makes sense as well.

Reply:
@eugene-manuilov I've updated the PR. Let me know what you think.